save-frontend/src/main/kotlin/com/saveourtool/save/frontend/App.kt

```diff
@@ -163,6 +163,7 @@ class App : ComponentWithScope<PropsWithChildren, AppState>() {
         HashRouter {
             requestModalHandler {
                 userInfo = state.userInfo
+
```

Comment:
I tried to tinker with your PR a little and found that this solution doesn't lead to endless loop:
```kotlin
                withRouter<Props> { location, _ ->
                    if (state.userInfo?.isActive == false && !location.pathname.startsWith("/${FrontendRoutes.REGISTRATION.path}")) {
                        Navigate {
                            to = "/${FrontendRoutes.REGISTRATION.path}"
                            replace = false
                        }
                    }
                }()
```

Basically, while `Navigate` is mounted in components tree, during each render it leads to another redirect. I'm still not sure, but I think that actual `window.location` and internal location representation in react-router's state are not fully synced. In my example we are asserting on location from `HashRouter`'s state.